apps/prairielearn/src/tests/sharedQuestionPreview.test.ts

```diff
@@ -0,0 +1,105 @@
+import { z } from 'zod';
+
+import {
+  testElementClientFiles,
+  testQuestionPreviews,
+  testFileDownloads,
+} from './helperQuestionPreview.js';
+import { config } from '../lib/config.js';
+import { features } from '../lib/features/index.js';
+import * as helperServer from './helperServer.js';
+import * as syncUtil from './sync/util.js';
+import * as sqldb from '@prairielearn/postgres';
+const sql = sqldb.loadSqlEquiv(import.meta.url);
+
+const siteUrl = 'http://localhost:' + config.serverPort;
+const baseUrl = siteUrl + '/pl';
+
+const addNumbers = {
+  id: '',
+  qid: 'addNumbers',
+  type: 'Freeform',
+  title: 'Add two numbers',
+};
+const addVectors = {
+  id: '',
+  qid: 'addVectors',
+  type: 'Calculation',
+  title: 'Addition of vectors in Cartesian coordinates',
+};
+const downloadFile = {
+  id: '',
+  qid: 'downloadFile',
+  type: 'Freeform',
+  title: 'File download example question',
+};
+const customElement = {
+  id: '',
+  qid: 'customElement',
+  type: 'Freeform',
+  title: 'Demo: Custom element',
+};
+const testQuestions = [addNumbers, addVectors, downloadFile, customElement];
+
+describe('Shared Question Preview', function () {
+  this.timeout(60000);
+
+  before('set up testing server', helperServer.before());
+  after('shut down testing server', helperServer.after);
+
+  before('ensure course has question sharing enabled', async () => {
+    await features.enable('question-sharing');
+  });
+
+  before('Get question IDs from database', async function () {
+    for (const testQuestion of testQuestions) {
+      testQuestion.id = await sqldb.queryRow(
+        sql.select_question_id,
+        {
+          qid: testQuestion.qid,
+        },
+        z.string(),
+      );
+      await sqldb.queryAsync(sql.update_shared_publicly, { question_id: testQuestion.id });
+    }
+  });
+
+  before('set up another course to consume shared questions from ', async () => {
+    const consumingCourseData = syncUtil.getCourseData();
+    consumingCourseData.course.name = 'CONSUMING 101';
+    await syncUtil.writeAndSyncCourseData(consumingCourseData);
+  });
+
+  describe('Test Public Question Previews', function () {
```

Comment:
Nit, this would better match our conventions (same for other tests below):

```suggestion
  describe('Public question previews', function () {
```

Reply:
done.